tests/unit-tests/test-class-course.php

```diff
@@ -549,4 +549,46 @@ public function data_testCourseArchiveOrderSetOrderBy() {
 			'Order set and default option selected'      => array( array( 'course-orderby' => 'default' ), 'anything', 'menu_order', 'ASC' ),
 		);
 	}
+
+	public function testGetAverageDaysToCompletionTotalReturnsMatchingValue() {
+		$user1_id  = $this->factory->user->create();
+		$user2_id  = $this->factory->user->create();
+		$user3_id  = $this->factory->user->create();
+		$course_id = $this->factory->course->create();
+
+		$comment1_id = Sensei_Utils::update_course_status( $user1_id, $course_id, 'complete' );
+		wp_update_comment(
+			[
+				'comment_ID'   => $comment1_id,
+				'comment_date' => '2022-01-07 00:00:00',
+			]
+		);
+		update_comment_meta( $comment1_id, 'start', '2022-01-01 00:00:01' );
+
+		$comment2_id = Sensei_Utils::update_course_status( $user2_id, $course_id, 'complete' );
+		wp_update_comment(
+			[
+				'comment_ID'   => $comment2_id,
+				'comment_date' => '2022-01-10 00:00:00',
+			]
+		);
+		update_comment_meta( $comment2_id, 'start', '2022-01-01 00:00:01' );
+
+		$comment3_id = Sensei_Utils::update_course_status( $user3_id, $course_id, 'complete' );
+		wp_update_comment(
+			[
+				'comment_ID'   => $comment3_id,
+				'comment_date' => '2022-01-30 00:00:00',
+			]
+		);
+		update_comment_meta( $comment3_id, 'start', '2022-01-01 00:00:01' );
+
+		$actual = Sensei()->course->get_average_days_to_completion_total();
```

Comment:
I don't like this approach of using shared code as it might have side-effects (as we share its state among all tests).
However, creating a new instance of `Sensei_Course` leads to longer tests running.
I suppose because of actions/filter.